packages/react/src/tooltip/trigger/TooltipTrigger.tsx

```diff
@@ -46,14 +46,18 @@ namespace TooltipTrigger {
     open: boolean;
   }
 
-  export interface Props extends BaseUIComponentProps<any, State> {}
+  export interface Props extends BaseUIComponentProps<'button', State> {}
```

Comment:
Tooltips can go on any type of element, not just buttons. It should at least have the `props: GenericHTMLProps` type instead of `props: any` though

Reply:
I don't know how to fix it cleanly, I have reverted TooltipTrigger fix.

Reply:
Thanks for pointing this out. I've made an extra PR here: https://github.com/mui/base-ui/pull/1213